pkg/kubelet/kubelet.go

```diff
@@ -3671,6 +3671,14 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *api.Pod, podStatus *kubeco
 
 	// Handle the containers failed to be started, which should be in Waiting state.
 	for _, container := range containers {
+		if hasInitContainers {
+			// If the init container is terminated with exit code 0, it won't be restarted.
+			// TODO(random-liu): Handle this in a cleaner way and add unit test for this.
```

Comment:
Could you add a simple unit test for init containers in this PR? At least we can make sure the fix works :)
The test should be similar to the `TestGenerateAPIPodStatus...` tests.